packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -146,25 +172,30 @@ class TreeTable<T extends TreeNode<T>> extends StatefulWidget {
   /// Factory that creates keys for each row in this table.
   final Key Function(T) keyFactory;
 
+  final ColumnData<T> startingSortColumn;
+
+  final SortDirection startingSortDirection;
+
   @override
   _TreeTableState<T> createState() => _TreeTableState<T>();
 }
 
 class _TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
-    with TickerProviderStateMixin {
+    with TickerProviderStateMixin, TableSortMixin<T> {
+  /// The number of items to show when animating out the tree table.
+  static const itemsToShowWhenAnimating = 50;
+
   List<T> items;
   List<T> animatingChildren = [];
   Set<T> animatingChildrenSet = {};
   T animatingNode;
   List<double> columnWidths;
   List<bool> rootsExpanded;
 
-  /// The number of items to show when animating out the tree table.
-  static const itemsToShowWhenAnimating = 50;
-
   @override
   void initState() {
     super.initState();
+    sortData(widget.startingSortColumn, widget.startingSortDirection);
```

Comment:
cleanup the code that is duplicated in didUpdateWidget and initState. add a helper called in both places.